### DISCUSSION TITLE
Improvements to the .NET BCL for wrappers

### DISCUSSION THREAD

**Discussion by jarle-moe** [MapsterMapper/Mapster#696] [General]
Hi,

I made an API proposal for the .NET BCL yesterday and in retrospect I was probably a bit naive in my approach.
Still, I'd be happy for some feedback on it..

If it's just a terrible idea in general, I'd also accept that 😅

https://github.com/dotnet/runtime/issues/101023

TLDR; I'd like the standard library to have some semantic understanding of wrappers; short example :
```csharp
public class MyWrapper<T>
{
    public T Value { get; set; }
}
```

In the case of Mapster, my thought is that the library would automatically understand that what it should do with that type is copy the wrapped value, and not the enclosing object. In other words, instead of `Id = 42` it should recognize it as just `42`, without individual programmers needing to implement this logic manually

**Comment by DocSvartz** [comment]
@jarle-moe in mapster only Value will be copied

```

[TestMethod]
public void Wrapper()
{
    var source = new WrapperInsaider { Number = new MyWrapper<int> { Value = 42 } };
    var target = new WrapperInsaider { Number = new MyWrapper<int> { Value = 101 } };

    var result = source.Adapt(target);

    object.Equals(source.Number, result.Number).ShouldBeFalse(); // it is not reference copy of source 
}
```
